### PR TITLE
Yang-models interface name length validation

### PR DESCRIPTION
**Pull request by arfeigin**
This PR is accompanied by https://github.com/sonic-net/sonic-swss-common/pull/889 and https://github.com/sonic-net/sonic-utilities/pull/3397

#### Why I did it
Validate interface name length does not exceed the limitation of IFNAMSIZ to align to kernel restrictions.
##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Add validation checks in the relevant yang-model definitions for the following interface types:

vxlan
vlan
vrf
loopback
subinterface
portchannel
#### How to verify it
UT added

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by nazariig — APPROVED** [review]

File: src/sonic-yang-models/yang-templates/sonic-types.yang.j2 [hunk 1]
```diff
@@ -316,6 +316,16 @@ module sonic-types {
         }
     }
 
+    typedef interface_name {
+        description "Represents IFNAMSIZ defined in <net/if.h>";
+        type string {
+            length 1..16 {
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-templates/sonic-types.yang.j2:322]: 16->15?
ref: https://stackoverflow.com/a/29398765/2514803

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**